File: ejercicios terminados/Java/estructuras repetitivas/act14.java
/*Programa: act14.java
 * 
 *Propósito: Realizar un programa que compruebe si una cadena contiene una subcadena. Las dos cadenas se introducen por teclado.
 * 
 *Autor: Jose Manuel Serrano Palomo.
 * 
 *Fecha: 10/11/2019
 *
 *Análisis:
 *
 *Diseño:
*/
import java.util.Scanner;
public class act14 {
    public static void main(String[] args) {
        System.out.println("Programa que realiza comprobacion de subcadena");
        System.out.println("----------------------------------------------act1");
        Scanner s = new Scanner(System.in);
        String cadena;
        String subcadena;
        boolean estaSubcadena = false;
        int numSubcadenas;

        System.out.print("Introduce una cadena: ");
        cadena = s.nextLine();
        System.out.print("Introduce una subcadena a buscar en la cadena anterior: ");
        subcadena = s.nextLine();

        numSubcadenas = cadena.length()-subcadena.length()+1;
        for (int nsubc=0; nsubc<=numSubcadenas-1 && !estaSubcadena; nsubc++) {
          if (cadena.substring(nsubc,nsubc+subcadena.length()).equals(subcadena)) {
            estaSubcadena = true;
          }
        }
        if (estaSubcadena) {
          System.out.println("La cadena contiene la subcadena.");
        } else {
          System.out.println("La cadena no contiene la subcadena.");
        }
    }
}

File: ejercicios terminados/Python/ejercicios repeticiones/act14.py
#Programa: act14.py
#Propósito: Realizar un programa que compruebe si una cadena contiene una subcadena. Las dos cadenas se introducen por teclado.
#Autor: Jose Manuel Serrano Palomo.
#Fecha: 17/10/2019
#
# Análisis:
# Leemos dos cadenas, hay que ir comprobando las subcadenas de la primera 
# que sean igual de tamaño que la segunda. Si alguna es igual, la primera contiene la segunda.
# Diseño:
# 

print("Añadimos dos cadenas y comprobamos si la segunda se encuentra en la primera")
print("------------------------------------------------------------------------------\n")

cad = str(input("Introduce una cadena: "))
subcad = str(input("Introduce una subcadena: "))
num_subcad = len(cad)-len(subcad) + 1
cad = list(cad)
ind = False

for i in range(0, num_subcad - 1):
    if num_subcad == subcad:
        ind = True
        
if ind:
    print("La cadena contiene la subcadena")
else:
    print("La cadena no tiene la subcadena")
